apache/server_status.sls

```diff
@@ -18,3 +18,25 @@ include:
       - module: apache-restart
       - module: apache-reload
       - service: apache
+
+{% if grains['os_family']=="Debian" %}
+a2endisconf server-status:
+  cmd.run:
+{%   if apache.get('server_status_require') is defined %}
+    - name: a2enconf server-status
+    - unless: test -L /etc/apache2/conf-enabled/server-status.conf
+{%   else %}
+    - name: a2disconf server-status
+    - onlyif: test -L /etc/apache2/conf-enabled/server-status.conf
+{%   endif %}
```

Comment:
As a matter of style, is it possible to declare some variable to [avoid putting jinja conditionals within the states as much as possible](https://docs.saltstack.com/en/latest/topics/development/conventions/formulas.html#light-conditionals-and-looping)?

For example:

``` jinja
{%- if grains['os_family']=="Debian" %}
{%-   if apache.get('server_status_require') is defined %}
{%-     set cmd = 'a2enconf server-status' %}
{%-     set cond = {'unless': 'test -L /etc/apache2/conf-enabled/server-status.conf'} %}
{%-   else %}
{%-     set cmd = 'a2disconf server-status' %}
{%-     set cond = {'onlyif': 'test -L /etc/apache2/conf-enabled/server-status.conf'} %}
{%-   endif %}
a2endisconf server-status:
  cmd.run:
    - name: {{ cmd }}
    - {{ cond }}
```

Reply:
> I'm fine with the changes.
> 
> I'm wondering what the `if apache.get('server_status_require') is defined` does when there is no `apache:server_status_require`, does the `is defined` handle the `KeyError`?

It does. Anyway, will refactor the code and add some tests also.

Reply:
> It does. Anyway, will refactor the code and add some tests also.

What do you think about this instead:

```jinja
{%-   if apache.get('server_status_require', False) %}
```

Otherwise, if the user set `apache:server_status_require` to `false` it will apply the configuration regardless.

Reply:
Actually, checking that logic, I removed that piece of code because it's wrong: as is, it would **disable** `server-status` when no _`server_status_require` entries_ are set. But the module, by default, is usually enabled with _localhost_ allowed and therefore these entries are optional.